sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/ScoringParameter.java

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.models;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a parameter value to be used in scoring functions (for example, referencePointParameter).
+ */
+public final class ScoringParameter {
+    private final ClientLogger logger = new ClientLogger(ScoringParameter.class);
+    private final String name;
+    private final List<String> values;
+
+    private static final String SEPARATOR = "-";
+    private static final String COMMA = ",";
+    private static final String SINGLE_QUOTE = "'";
+    /**
+     * Initializes a new instance of the ScoringParameter class with the given name and string values.
+     *
+     * @param name Name of the scoring parameter.
+     * @param values Values of the scoring parameter.
+     * @throws NullPointerException if {@code name} or {@code values} is null.
+     */
+    @JsonCreator
+    public ScoringParameter(@JsonProperty(value = "name") String name,
+        @JsonProperty(value = "values") List<String> values) {
+        Objects.requireNonNull(name);
+        Objects.requireNonNull(values);
+        this.name = name;
+        this.values = new ArrayList<>();
+        this.values.addAll(values);
+    }
+
+    /**
+     * Initializes a new instance of the ScoringParameter class with the given name and GeographyPoint value.
+     *
+     * @param name Name of the scoring parameter.
+     * @param value Value of the scoring parameter.
+     */
+    public ScoringParameter(String name, GeoPoint value) {
+        this(name, toLonLatStrings(value));
+    }
+
+    /**
+     * Gets the name of the scoring parameter.
+     *
+     * @return The name of scoring parameter.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Gets the values of the scoring parameter.
+     *
+     * @return The values of scoring parameter.
+     */
+    public List<String> getValues() {
+        return values;
+    }
+
+    private static List<String> toLonLatStrings(GeoPoint point) {
+        if (point == null) {
+            return null;
+        }
```

Comment:
Should this just be `Objects.requireNonNull`, if `null` is returned here it will eventually lead to a `NullPointerException` in another, less obvious, location.

Reply:
I planned to remove the null checking here.
Since there are two constructors in this class. User cannot put null to initialize the constructor, it will throw compile time error.

Reply:
switch to null pointer exception check.